src/In.ProjectEKA.TMHHip/DataFlow/Collect.cs

```diff
@@ -0,0 +1,752 @@
+                                careBundles.Add(careBundle);
+                            }
+                        }
+
+                        break;
+                    }
+                    case HiType.DiagnosticReport:
+                    {
+                        if (tmhPatientData.DiagnosticReportAsPdfs != null &&
+                            tmhPatientData.DiagnosticReportAsPdfs.Any())
+                        {
+                            var diagnosticReportResponse = FetchDiagnosticReportPdfsData(dataRequest,
+                                tmhPatientData.DiagnosticReportAsPdfs, patientName).GetAwaiter().GetResult();
+                            if (diagnosticReportResponse.HasValue)
+                            {
+                                var serializeObject = JsonConvert.SerializeObject(
+                                    diagnosticReportResponse.ValueOr((DiagnosticReportResponse) null),
+                                    new JsonSerializerSettings
+                                    {
+                                        Formatting = Formatting.Indented,
+                                        NullValueHandling = NullValueHandling.Ignore,
+                                        ContractResolver = new CamelCasePropertyNamesContractResolver()
+                                    });
+                                var careBundle = new CareBundle(caseId, parser.Parse<Bundle>(serializeObject));
+                                careBundles.Add(careBundle);
+                            }
+                        }
+
+                        if (tmhPatientData.DiagnosticReportAsImages != null &&
+                            tmhPatientData.DiagnosticReportAsImages.Any())
+                        {
+                            var diagnosticReportResponse = FetchDiagnosticReportImagesData(dataRequest,
+                                tmhPatientData.DiagnosticReportAsImages, patientName).GetAwaiter().GetResult();
+                            if (diagnosticReportResponse.HasValue)
+                            {
+                                var serializeObject = JsonConvert.SerializeObject(
+                                    diagnosticReportResponse.ValueOr((DiagnosticReportResponse) null),
+                                    new JsonSerializerSettings
+                                    {
+                                        Formatting = Formatting.Indented,
+                                        ContractResolver = new CamelCasePropertyNamesContractResolver()
+                                    });
+                                var careBundle = new CareBundle(caseId, parser.Parse<Bundle>(serializeObject));
+                                careBundles.Add(careBundle);
+                            }
+                        }
+
+                        careBundles.Add(new CareBundle(caseId, await ReadJsonAsync<Bundle>("MockedDicomImageData.json")));
```

Comment:
Codacy found an issue: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=4940156)